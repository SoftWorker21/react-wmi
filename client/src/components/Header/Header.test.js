import { render, screen } from "@testing-library/react";
import { Header } from "./index";
import data from "../../honda_wmi.json";

test("render Header react link", () => {
  render(<Header />);
  const linkElement = screen.getByText(/WMI Data - Honda/i);
  expect(linkElement).toBeInTheDocument();
});

test("render Header car data prop", () => {
  render(<Header carData={data} />);
  const linkElement = screen.getByText(/Total: 82/i);
  expect(linkElement).toBeInTheDocument();
});

test("render Header input", () => {
  render(<Header carData={data} />);
  const linkElement = screen.getByPlaceholderText(/What you looking for?/i);
  expect(linkElement).toBeInTheDocument();
});

test("render Header snapshot", () => {
  const tree = render(<Header carData={data} />);
  expect(tree).toMatchSnapshot();
});
