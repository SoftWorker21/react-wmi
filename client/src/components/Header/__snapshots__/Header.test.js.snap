// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h2>
          WMI Data - Honda | Total: 
          82
        </h2>
        <div
          class="filter-box"
        >
          <p>
             items
          </p>
          <input
            placeholder="What you looking for?"
            type="text"
            value=""
          />
          <select
            data-testid="select"
            id="cars"
            name="cars"
          />
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <h2>
        WMI Data - Honda | Total: 
        82
      </h2>
      <div
        class="filter-box"
      >
        <p>
           items
        </p>
        <input
          placeholder="What you looking for?"
          type="text"
          value=""
        />
        <select
          data-testid="select"
          id="cars"
          name="cars"
        />
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
